### DISCUSSION TITLE
Enable cgroup v1

### DISCUSSION THREAD

**Discussion by jfernandez** [arighi/virtme-ng#84] [General]
I'm running vng from a Linux host running Ubuntu 22.04. I'd like to disable cgroup v2 to test a patch for cgroup v1. I've tried this and it didn't work: `vng --append "systemd.unified_cgroup_hierarchy=0"`

**Comment by arighi** [comment]
Hi @jfernandez , vng doesn't run systemd inside the guest (it's actually using a custom init, to speed up the boot time). But we can definitely mimic the same systemd behavior and mount cgroup v1 if this option is passed. I'll work on that later today.

**Comment by arighi** [comment]
From systemd's documentation:
```
CHANGES WITH 256 in spe:
...
        * Support for cgroup v1 ('legacy' and 'hybrid' hierarchies) is now
          considered obsolete and systemd by default will refuse to boot under it.
          To forcibly reenable cgroup v1 support, SYSTEMD_CGROUP_ENABLE_LEGACY_FORCE=1
          must be set on kernel command line.
```

I'm wondering if we should do the same in vng, instead of supporting the three modes of `systemd.unified_cgroup_hierarchy=[0|1|2]`.

Basically using `vng ... --append "SYSTEMD_CGROUP_ENABLE_LEGACY_FORCE=1"` would just mount cgroup v1 under `/sys/fs/cgroup` using the legacy layout (tmpfs + a separate mount-point for each cgroup v1 subsystem). This would allow to test cgroup v1 when needed, and I don't think anybody wants to test the hybrid mode anymore (cgroup v1 + v2 mounted at the same time).

@jfernandez would something like this work for you?

**Reply by jfernandez** [reply]
Yes it would. You are correct that I don’t need hybrid mode. I’m working on a BPF kfunc and I need to verify its behavior for cgroup v1.

**Comment by arighi** [comment]
@jfernandez can you give it a try with #85?

It seems to work for me:
```
$ vng -r --append "SYSTEMD_CGROUP_ENABLE_LEGACY_FORCE=1" -- 'df -T /sys/fs/cgroup/*'
Filesystem     Type   1K-blocks  Used Available Use% Mounted on
blkio          cgroup         0     0         0    - /sys/fs/cgroup/blkio
cpu            cgroup         0     0         0    - /sys/fs/cgroup/cpu
cpuacct        cgroup         0     0         0    - /sys/fs/cgroup/cpuacct
devices        cgroup         0     0         0    - /sys/fs/cgroup/devices
memory         cgroup         0     0         0    - /sys/fs/cgroup/memory
pids           cgroup         0     0         0    - /sys/fs/cgroup/pids
6.8.0-13-generic ~
$ vng -r -- 'df -T /sys/fs/cgroup/'
Filesystem     Type    1K-blocks  Used Available Use% Mounted on
cgroup2        cgroup2         0     0         0    - /sys/fs/cgroup
```

**Reply by jfernandez** [reply]
I’ll give it a try by EOD. Thank so much for the quick turnaround time.
